framework/src/materials/CoupledGradientMaterial.C

```diff
@@ -0,0 +1,50 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "CoupledGradientMaterial.h"
+
+registerMooseObject("MooseApp", CoupledGradientMaterial);
+registerMooseObject("MooseApp", ADCoupledGradientMaterial);
+
+template <bool is_ad>
+InputParameters
+CoupledGradientMaterialTempl<is_ad>::validParams()
+{
+  InputParameters params = Material::validParams();
+  params.addClassDescription("Create a gradient material coupled to a variable.");
+  params.addRequiredParam<MaterialPropertyName>(
+      "grad_mat_prop",
+      "Name of gradient material property equal to the gradient of the coupled variable gradient "
+      "times the scalar.");
+  params.deprecateParam("grad_mat_prop", "gradient_material_name", "12/12/25");
+  params.addParam<MaterialPropertyName>(
+      "scalar_property",
```

Comment:
I d deprecate this name and call it scalar_property_factor

Reply:
No need to deprecate this one, it is new. I'll change the name.